app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/CustomizeHomeButtonViewHolder.kt

```diff
@@ -37,12 +47,44 @@ class CustomizeHomeButtonViewHolder(
 
     @Composable
     override fun Content() {
+        val wallpaperState = components.appStore
+            .observeAsComposableState { state -> state.wallpaperState }.value ?: WallpaperState.default
+        val wallpaperIsDefault = Wallpaper.nameIsDefault(wallpaperState.currentWallpaper.name)
+        val buttonColor: Color
+        val textColor: Color
+        when {
+            !wallpaperIsDefault &&
+                wallpaperState.currentWallpaper.cardColorLight != null &&
+                wallpaperState.currentWallpaper.cardColorDark != null -> {
+                if (isSystemInDarkTheme()) {
+                    buttonColor = Color(wallpaperState.currentWallpaper.cardColorLight)
+                    textColor = FirefoxTheme.colors.textPrimary
+                } else {
+                    buttonColor = Color(wallpaperState.currentWallpaper.cardColorDark)
+                    textColor = FirefoxTheme.colors.textActionPrimary
+                }
+            }
+            else -> {
+                buttonColor = FirefoxTheme.colors.actionTertiary
+                textColor = FirefoxTheme.colors.textActionTertiary
+            }
+        }
```

Comment:
We can simplify this by reducing the number of lines by setting the default colors ahead of time and remove the `else`  condition.

Reply:
This is common theme in this PR. I would consider this idea holistically as you go back in this PR.